### ISSUE TITLE
Change of NWS products

### ISSUE THREAD

**Issue by Sudo-Rob** [MatthewFlamm/nwsradar#16]
This is FYI.

It looks as if the NWS will be rolling out a new service on December 8th.  Here is a preview of the new site:

[https://preview-radar.weather.gov/](url)

I know the current local websites have not been maintained for a while (and likely won't be) in anticipation of the debut of the new services.   What's unknown is if the current NEXRAD products will be deprecated.

**Comment by MatthewFlamm** [comment]
Hmmm...  I'll cross my fingers that the current images will still exist, while I look into this new system.

Thanks for the report!

**Comment by Sudo-Rob** [comment]
It looks like this might be the new repository:

[https://preview-opengeo.ncep.noaa.gov/geoserver/www/index.html](url)

**Comment by MatthewFlamm** [comment]
Your links are weird in that they point to https://github.com/MatthewFlamm/nwsradar/issues/url

Anyway, OWSLib  https://github.com/geopython/OWSLib looks like it is built to parse this API.  As far as I can tell, it won't be easy to install for use in home assistant.  pyproj is a dependency, and it requires PROJ be installed which cannot be installed via `pip`.

**Comment by Sudo-Rob** [comment]
Yes, I saw that.  I just used the link icon above this 'write' window.  

I just looked at the repository.  It seems to be all metadata.

If you need a contact at NOAA, I've exchanged e-mails with someone who might be able to point you to the group supporting API access.  I realize it's a longshot given the anticipated number of users of your integration.  I don't wish to post the address here, however.

**Comment by MatthewFlamm** [comment]
I think I am able to get the map layers like reflectivity from the web map service, but it looks like they no longer provide the base maps in the same server.  This makes the whole thing way more complicated than the current implementation unfortunately.  It looks like there are other servers from usgs that could provide base maps, but again, this makes it an order of magnitude more complex.

**Comment by Sudo-Rob** [comment]
Sorry to hear of the complications.  I wonder if there's any open-source code for combining the products, or an existing widget for embedment.

In the meantime, I'll keep my fingers crossed the existing maps won't be deprecated.

**Comment by Sudo-Rob** [comment]
This link might be helpful.  The link seems to work properly when I don't use the link icon.

https://www.ncdc.noaa.gov/data-access/radar-data/radar-decoding

**Comment by jaydenthorup** [comment]
Well they swapped it today

**Comment by jaydenthorup** [comment]
I am working around this by just using a webpage card and grabbing the bookmark per radar.weather.gov:

**Bookmarks
The URL will automatically update as you select the view and settings. You may bookmark the URL to return later to the same view with the selected settings. For example, if you select "Weather for a location," then select a location, the bookmark will return to your location on your next visit.

**Comment by MatthewFlamm** [comment]
So is the current integration totally not working?  I am away from home and cannot check easily.  I have a proof of concept script that uses the new api.  Unfortunately the new api server was down for 2-3 week, so I haven't gotten too far.  It will be awhile until I can get it to be generally usable.

**Comment by Sudo-Rob** [comment]
Yes, I'm seeing a stream of Python errors in the log.  The LWX NOAA site is not quite functional, but they're linking directions for using the new products.

https://www.weather.gov/ctp/newRadarTutorial

**Comment by Sudo-Rob** [comment]
> The URL will automatically update as you select the view and settings. You may bookmark the URL to return later to the same view with the selected settings. For example, if you select "Weather for a location," then select a location, the bookmark will return to your location on your next visit.

Unless I'm doing something wrong, 1/3rd of the iframe is banner and menu.  Is there any better view I'm missing?

**Comment by MatthewFlamm** [comment]
If the old server doesn't come back in a few days, I will probably remove this integration from HACS and archive it here.  A new integration will likely require different configuration anyway.

**Comment by Sudo-Rob** [comment]
Understand.  Thanks for what you've done, and good luck with any new integration.

**Comment by Sudo-Rob** [comment]
I've found this repository:

https://www.weather.gov/radarfaq
https://mrms.ncep.noaa.gov/data/RIDGEII/L3/

It looks like there are current files in gzip format.  Unfortunately, I haven't been able to extract any of the contents to see what products are in them.  I keep getting an 'invalid tar magic' diagnostic.  You can drill down into the gzip files, e.g. https://mrms.ncep.noaa.gov/data/RIDGEII/L3/KLWX/BSTP/

I hope this stuff turns out to be useful.

**Comment by gdreelin** [comment]
Looks like what every they did broke the ability to see the radar now, just noticed it was no showing anymore. I get an unresponsive error on it now. I tried using the full name of the station KFCX and then tried dropping the "k" like some other ones do to FCX. No joy on either.

**Comment by MatthewFlamm** [comment]
This integration is most likely dead.  I have a PR to remove from HACS default.  I think you will have to remove the integration from home assistant manually.

https://github.com/hacs/default/pull/747
